Extensions/Content/Dnn.PersonaBar.Pages/Pages.Web/src/actions/pageActions.js

```diff
@@ -206,26 +206,30 @@ const pageActions = {
         };
     },
 
-    deletePage(page, redirectUrl) {
+    deletePage(page, hardDelete, redirectUrl, callback) {
+        
         return (dispatch) => {
             dispatch({
                 type: ActionTypes.DELETE_PAGE
             });
 
-            PagesService.deletePage(page).then(response => {
+            PagesService.deletePage(page, hardDelete).then(response => {
 
                 if (response.Status === responseStatus.ERROR) {
                     utils.notifyError(response.Message, 3000);
                     return;
                 }
 
-                dispatch({
-                    type: ActionTypes.DELETED_PAGE
-                });
-                if (page.tabId !== 0 && (page.tabId === utils.getCurrentPageId()) || redirectUrl) {
-                    window.top.location.href = redirectUrl ? redirectUrl : utils.getDefaultPageUrl();
+                if(typeof callback === "function"){
+                    callback();
```

Comment:
Using `Promise` or even `async/await` would be better than adding a callback parameter

Reply:
In which case maybe this should be another `dispatch` instead?